insonmnia/miner/network/l2tp_state.go

```diff
@@ -0,0 +1,267 @@
+package network
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/docker/libkv"
+	"github.com/docker/libkv/store"
+	"github.com/docker/libkv/store/boltdb"
+	log "github.com/noxiouz/zapctx/ctxlog"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	pppOptsDir = "/etc/ppp/"
+)
+
+type l2tpState struct {
+	Aliases  map[string]string
+	Networks map[string]*l2tpNetwork
+	storage  store.Store
+	logger   *zap.SugaredLogger
+}
+
+func newL2TPNetworkState(ctx context.Context, path string) (*l2tpState, error) {
+	boltdb.Register()
+	var (
+		backend   = store.Backend(store.BOLTDB)
+		endpoints = []string{path}
+		config    = store.Config{Bucket: "sonm_l2tp_driver_state"}
+	)
+	storage, err := libkv.NewStore(backend, endpoints, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	state := &l2tpState{
+		Aliases:  make(map[string]string),
+		Networks: make(map[string]*l2tpNetwork),
+		logger:   log.S(ctx).With("source", "l2tp/store"),
+		storage:  storage,
+	}
+
+	if err := state.load(); err != nil {
+		return nil, err
+	}
+
+	return state, nil
+}
+
+func (s *l2tpState) Sync() (err error) {
+	defer func() {
```

Comment:
I'd rather instead make `sync() error` with the logic inside and `Sync` with error checking and logging.